### PR TITLE
[CFM-1194] Add group access behat tests (Re-rolled).

### PR DESCRIPTION
**Pull request by ybaras**
https://issuetracker.amplexor.com/jira/browse/CFM-1194

### CODE REVIEW

File: project/profiles/capacity4more/behat/features/group_content_access.feature [hunk 1]
```diff
@@ -35,3 +35,12 @@ Feature: Group content access
     When  I am logged in as user "isaacnewton"
     Then  I visit "Test content" node of type "discussion"
     And   I should see "Access denied"
+
+  @javascript
+  Scenario: A member of the organization should be able to see the group's content.
+    Given I am logged in as user "turing"
+    And   I change access of group "My test group" to "Restricted" to domain "turingmachine.com"
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: Replace `to domain` => `with the domain`

File: project/profiles/capacity4more/behat/features/group_content_access.feature [hunk 2]
```diff
@@ -35,3 +35,12 @@ Feature: Group content access
...
+    When  I am logged in as user "charlesbabbage"
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: Shouldn't we log out before and make sure we don't have access to the page ?

File: project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/csv/user/user.csv [hunk 3]
```diff
@@ -7,3 +7,4 @@ user_5,survivalofthefittest,drupal,charles.darwin@evolution.com,Charles,Darwin,"
...
 user_7,president,drupal,jean@ec.europa.eu,Jean-Claude,Juncker,"The President is the head of the European Commission. According to the Treaties, he decides on the organisation of the Commission, allocates portfolios to members of the Commission and can make changes at any time.",jean.jpg,European Commission,european_institutions,LU,,,,
 user_8,alfrednobel,drupal,alfred@nobel.se,Alfred,Nobel,"He was the inventor of dynamite. His fortune was used posthumously to institute the Nobel Prizes.",alfrednobel.jpg,Nobelprize,private_sector,SE,t_wind|t_earth,t_fire,Sweden|Worldwide,
+user_9,charlesbabbage,drupal,charlie@turingmachine.com,Charles,Babbage,"A mathematician, philosopher, inventor and mechanical engineer, who is best remembered for originating the concept of a programmable computer.",Charles_Babbage_-_1860.jpg,Turing Testing Inc.,government,GB,,,United Kingdom,
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/demo/c4m_demo_content/csv/user/user.csv]: Change the image file name to _charlesbabbage.jpg_ (just to keep the convention).

File: project/profiles/capacity4more/behat/features/group_content_access.feature [hunk 4]
```diff
@@ -35,3 +35,21 @@ Feature: Group content access
...
+    And   I change access of group "My test group" to "Restricted" with the domain "turingmachine.com"
+    When  I am an anonymous user
+    Then  I visit "Test content" node of type "discussion"
+    And   I should not see "Test content"
+    And   I should see "Please log in to continue"
+
+  @javascript
+  Scenario: A member of the organization should be able to see the group's content.
+    Given I am logged in as user "turing"
+    And   I change access of group "My test group" to "Restricted" with the domain "turingmachine.com"
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: @ybaras 
You already invoked the same command at the previous scenario, hence the access expected to be like that already, so try to remove this line.

File: project/profiles/capacity4more/behat/features/group_content_access.feature [hunk 5]
```diff
@@ -35,3 +35,21 @@ Feature: Group content access
...
+    When  I am logged in as user "charlesbabbage"
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: I guess the scenario should start from here.

Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: Of course replace the `When` with `Given`

File: project/profiles/capacity4more/behat/features/group_content_access.feature [hunk 6]
```diff
@@ -35,3 +35,19 @@ Feature: Group content access
     When  I am logged in as user "isaacnewton"
     Then  I visit "Test content" node of type "discussion"
     And   I should see "Access denied"
+
+  @javascript
```
Comment (amitaibu) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: why not `@api`?

File: project/profiles/capacity4more/behat/features/group_content_access.feature [hunk 7]
```diff
@@ -35,3 +35,19 @@ Feature: Group content access
...
+  Scenario: An anonymous user shouldn't be able to see a restricted group's content.
+    Given I am logged in as user "turing"
+    And   I change access of group "My test group" to "Restricted" with the domain "turingmachine.com"
+    When  I am an anonymous user
+    Then  I visit "Test content" node of type "discussion"
+    And   I should not see "Test content"
```
Comment (amitaibu) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: it's better to check access denied, no need to check for "login"

File: project/profiles/capacity4more/behat/features/group_content_access.feature (hunk 6 above)
Comment (amitaibu) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: is it because you need to JS to create the content?

Comment (ybaras) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: Yes, the 1st test in the feature creates a test group and adds content.

File: project/profiles/capacity4more/behat/features/group_content_access.feature (hunk 7 above)
Comment (ybaras) [inline project/profiles/capacity4more/behat/features/group_content_access.feature]: Anon users are redirected to the login page. I will add another test to check for `Access Denied` If a user **is** logged in but not authorized to view the content.

**Comment by soniCaH** [comment]
@ybaras can you delete the other branch for this ticket if it is obsolete please? ({{CFM-1194-group-access-behat}})

**Pull request merged**